src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import styles from './PuzzleGenerate.module.scss';
+import { Button, Input } from '../../Global';
+import { createPuzzleInstance, getAllPuzzles } from '../../../services';
+
+const PuzzleGenerate = () => {
+  const [puzzleList, setPuzzleList] = useState([]);
+  const [hint, setHint] = useState('');
+  const [latitude, setLatitude] = useState();
+  const [longitude, setLongitude] = useState();
+  const [address, setAddress] = useState('');
+  const [puzzleId, setPuzzleId] = useState('');
+
+  const handleSubmit = () => {
+    console.log(hint, latitude, longitude, address, puzzleId);
+    puzzleId &&
+      createPuzzleInstance(
+        parseInt(puzzleId),
+        longitude,
+        latitude,
+        address,
+        hint
+      ).then(puzzleInstance => console.log(puzzleInstance));
+  };
+
+  useEffect(() => {
+    const fetchPuzzles = async () => {
+      const puzzles = await getAllPuzzles();
+      setPuzzleList(puzzles);
+      puzzles.length > 0 && setPuzzleId(puzzles[0].id);
```

Comment:
But like, what happens if you just don't do the setPuzzleId part here? This feels bad because it means any time this useEffect is triggered, it will default the selection back to the first puzzle in the list, even if the user had something else selected, right?

Why do we need to set the puzzle id value here at all?

Reply:
This useEffect will only be triggered when you first load the page, don't worry. So initially, the puzzleId was null, but I designed the <option> to select the first puzzle as a default. This is like setting useState('') for textArea.

Reply:
Oh wait but then let's just do the call in `getServerSideProps()`. There's no need for a useEffect hook that's only called once when the page loads.

Just pass a list of puzzles as props to this page

Reply:
Woah I did not even know that exists 🤩